File: backend/src/main/java/com/cityvoice/auth/dto/TokenResponse.java
package com.cityvoice.auth.dto;

import com.cityvoice.user.enums.UserRole;

import java.util.UUID;

public record TokenResponse(
        String accessToken,
        String refreshToken,
        String tokenType,
        long accessExpiresIn) {
    public static TokenResponse of(String accessToken, String refreshToken, long accessExpiresInMs) {
        return new TokenResponse(accessToken, refreshToken, "Bearer", accessExpiresInMs / 1000);
    }
}

File: mobile/lib/features/auth/models/token_response.dart
/// Represents the token pair returned by login and refresh endpoints.
///
/// Maps to the backend `TokenResponse` record:
/// ```json
/// {
///   "accessToken": "...",
///   "refreshToken": "...",
///   "tokenType": "Bearer",
///   "accessExpiresIn": 9000
/// }
/// ```
class TokenResponse {
  final String accessToken;
  final String refreshToken;
  final String tokenType;
  final int accessExpiresIn;

  const TokenResponse({
    required this.accessToken,
    required this.refreshToken,
    this.tokenType = 'Bearer',
    this.accessExpiresIn = 0,
  });

  factory TokenResponse.fromJson(Map<String, dynamic> json) {
    return TokenResponse(
      accessToken: json['accessToken']?.toString() ?? '',
      refreshToken: json['refreshToken']?.toString() ?? '',
      tokenType: json['tokenType']?.toString() ?? 'Bearer',
      accessExpiresIn: (json['accessExpiresIn'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'accessToken': accessToken,
        'refreshToken': refreshToken,
        'tokenType': tokenType,
        'accessExpiresIn': accessExpiresIn,
      };
}
